src/Yasumi/Provider/Greece.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ *  This file is part of the Yasumi package.
+ *
+ *  Copyright (c) 2015 - 2016 AzuyaLabs
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ *  @author Sacha Telgenhof <stelgenhof@gmail.com>
+ */
+namespace Yasumi\Provider;
+
+use DateInterval;
+use DateTime;
+use DateTimeZone;
+use Yasumi\Holiday;
+
+/**
+ * Provider for all holidays in Greece.
+ */
+class Greece extends AbstractProvider
+{
+    use CommonHolidays, ChristianHolidays;
+
+    /**
+     * Initialize holidays for Greece.
+     */
+    public function initialize()
+    {
+        $this->timezone = 'Europe/Athens';
+
+        // Add common holidays
+        $this->addHoliday($this->newYearsDay($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->internationalWorkersDay($this->year, $this->timezone, $this->locale));
+
+
+        // Add Christian holidays
+        $this->addHoliday($this->epiphany($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->annunciation($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->goodFriday($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->easter($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->easterMonday($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->ascensionDay($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->pentecost($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->pentecostMonday($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->assumptionOfMary($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->christmasDay($this->year, $this->timezone, $this->locale));
+
+        // Calculate other holidays
+        $this->calculateThreeHolyHierarchs();
+        $this->calculateCleanMonday();
+        $this->calculateIndependenceDay();
+        $this->calculateOhiDay();
+        $this->calculatePolytechnio();
```

Comment:
Ochi day is a national non-working holiday. Polytechnio as well as The Three Hierarchs are working days, but schools and universities are closed.